### PR TITLE
Add JSDoc for accordion constructor

### PR DESCRIPTION
**Pull request by 36degrees**

### CODE REVIEW

File: src/govuk/components/accordion/accordion.mjs [hunk 1]
```diff
@@ -1,25 +1,32 @@
...
 import I18n from '../../i18n.mjs'
 import '../../vendor/polyfills/Function/prototype/bind.mjs'
 import '../../vendor/polyfills/Element/prototype/classList.mjs'
 
+/**
+ * Accordion component
+ *
+ * This allows a collection of sections to be collapsed by default, showing only
+ * their headers. Sections can be expanded or collapsed individually by clicking
+ * their headers. An "Show all sections" button is also added to the top of the
```
Comment (domoscargin) [inline src/govuk/components/accordion/accordion.mjs:11]:
```suggestion
 * their headers. A "Show all sections" button is also added to the top of the
```

File: src/govuk/components/accordion/accordion.mjs [hunk 2]
```diff
@@ -1,25 +1,32 @@
...
+ * accordion, which switches to "Hide all sections" when all the sections are
+ * expanded.
+ *
+ * The state of each section is saved to the DOM via the `aria-expanded`
+ * attribute, which also provides accessibility.
+ *
```
Comment (romaricpascal) [inline src/govuk/components/accordion/accordion.mjs:17]: Given the function is used as constructor, we should highlight it in the JSDoc as well.
```suggestion
 *
 * @class
```

File: src/govuk/components/accordion/accordion.mjs [hunk 3]
```diff
@@ -1,25 +1,32 @@
...
+ * @param {Object} config
+ * @param {Object} config.i18n - Translations
+ * @param {String} [config.i18n.hideAllSections='Hide all sections'] - Text for
+ *   'hide all sections' button, displayed when at least one section is expanded
+ * @param {String} [config.i18n.hideSection='Hide<span class="govuk-visually-hidden"> this section</span>']
+ *   - Text for 'hide this section' button, displayed when a section is expanded
+ * @param {String} [config.i18n.showAllSections='Show all sections'] - Text for
+ *   'show all sections' button, displayed when all sections are collapsed
+ * @param {String} [config.i18n.showSection='Show<span class="govuk-visually-hidden"> this section</span>']
+ *   - Text for 'show section' button, displayed when a section is collapsed
```
Comment (domoscargin) [inline src/govuk/components/accordion/accordion.mjs:28]:
```suggestion
 *   - Text for 'show this section' button, displayed when a section is collapsed
```

**Review by domoscargin — APPROVED** [review]
Changes look good to me!

**Pull request merged**